packages/firestore/src/local/local_store.ts

```diff
@@ -1389,21 +1417,41 @@ export async function saveNamedQuery(
     'readwrite',
     transaction => {
       // Update allocated target's read time, if the bundle's read time is newer.
-      let updateReadTime = PersistencePromise.resolve();
+      let updateReadTime = PersistencePromise.resolve(false);
       const readTime = fromVersion(query.readTime!);
       if (allocated.snapshotVersion.compareTo(readTime) < 0) {
         const newTargetData = allocated.withResumeToken(
           ByteString.EMPTY_BYTE_STRING,
           readTime
         );
-        updateReadTime = localStoreImpl.targetCache.updateTargetData(
-          transaction,
+        updateReadTime = localStoreImpl.targetCache
+          .updateTargetData(transaction, newTargetData)
+          .next(
+            () => true,
+            () => false
+          );
```

Comment:
This needs explanation (but I think it could be removed if you update the target mapping right here)